libsolidity/formal/SMTEncoder.cpp

```diff
@@ -730,6 +731,47 @@ void SMTEncoder::visitRequire(FunctionCall const& _funCall)
 	addPathImpliedExpression(expr(*args.front()));
 }
 
+void SMTEncoder::visitCryptoFunction(FunctionCall const& _funCall)
+{
+	FunctionType const& funType = dynamic_cast<FunctionType const&>(*_funCall.expression().annotation().type);
+	auto kind = funType.kind();
+	auto arg0 = expr(*_funCall.arguments().at(0));
+	optional<smtutil::Expression> result;
+	if (kind == FunctionType::Kind::KECCAK256)
+	{
+		auto k = m_context.state().cryptoFunction("keccak256");
+		result = smtutil::Expression::select(k, arg0);
```

Comment:
Can this be one line maybe?
```suggestion
		result = smtutil::Expression::select(m_context.state().cryptoFunction("keccak256"), arg0);
```

Reply:
Sure, I just thought it'd be easier to read by getting the array first, but fine with either.

Reply:
I would be fine with both as well